Services/Component/classes/Setup/class.ilComponentActivatePluginsObjective.php

```diff
@@ -0,0 +1,150 @@
+
+declare(strict_types=1);
+
+use ILIAS\Setup;
+use ILIAS\DI;
+
+class ilComponentActivatePluginsObjective implements Setup\Objective
+{
+    /**
+     * @var string
+     */
+    protected $plugin_name;
+
+    public function __construct(string $plugin_name)
+    {
+        $this->plugin_name = $plugin_name;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getHash() : string
+    {
+        return hash("sha256", self::class . $this->plugin_name);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getLabel() : string
+    {
+        return "Activate plugin $this->plugin_name.";
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function isNotable() : bool
+    {
+        return true;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getPreconditions(Setup\Environment $environment) : array
+    {
+        $setup_config = $environment->getConfigFor('common');
+        $db_config = $environment->getConfigFor('database');
+
+        return [
+            new \ilIniFilesPopulatedObjective($setup_config),
+            new \ilDatabasePopulatedObjective($db_config),
+            new \ilComponentPluginAdminInitObjective()
+        ];
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function achieve(Setup\Environment $environment) : Setup\Environment
+    {
+        $ORIG_DIC = $this->initEnvironment($environment);
+
+        $plugin = $GLOBALS["DIC"]["ilPluginAdmin"]->getRawPluginDataFor($this->plugin_name);
+
+        if (!is_null($plugin) && $plugin['activation_possible'] && $plugin['supports_cli_setup']) {
+            $pl = ilPlugin::getPluginObject(
+                $plugin['component_type'],
+                $plugin['component_name'],
+                $plugin['slot_id'],
+                $plugin['name']
+            );
+
+            $pl->activate();
+        }
+
+        $GLOBALS["DIC"] = $ORIG_DIC;
+
+        return $environment;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function isApplicable(Setup\Environment $environment) : bool
+    {
+        $ORIG_DIC = $this->initEnvironment($environment);
+
+        $plugin = $GLOBALS["DIC"]["ilPluginAdmin"]->getRawPluginDataFor($this->plugin_name);
+
+        if (is_null($plugin) || !$plugin['supports_cli_setup']) {
+            return false;
+        }
+
+        $GLOBALS["DIC"] = $ORIG_DIC;
+
+        return $plugin['activation_possible'];
+    }
+
+    protected function initEnvironment(Setup\Environment $environment) : ILIAS\DI\Container
+    {
+        $db = $environment->getResource(Setup\Environment::RESOURCE_DATABASE);
+        $plugin_admin = $environment->getResource(Setup\Environment::RESOURCE_PLUGIN_ADMIN);
+        $ini = $environment->getResource(Setup\Environment::RESOURCE_ILIAS_INI);
+        $client_ini = $environment->getResource(Setup\Environment::RESOURCE_CLIENT_INI);
+
+
+        // ATTENTION: This is a total abomination. It only exists to allow various
```

Comment:
@klees Thanks for the note here, this is absolutely correct and definitely something we need to tackle in the future.